lib/oli_web/live/new_course/select_source.ex

```diff
@@ -0,0 +1,268 @@
+defmodule OliWeb.Delivery.NewCourse.SelectSource do
+  use Surface.LiveComponent
+
+  alias Oli.Delivery
+  alias Oli.Delivery.Sections.Blueprint
+  alias Oli.Publishing
+  alias OliWeb.Common.{Filter, FilterBox, Listing, SessionContext}
+  alias OliWeb.Common.Table.SortableTableModel
+  alias OliWeb.Router.Helpers, as: Routes
+
+  alias Surface.Components.Form
+  alias Surface.Components.Form.{Field, RadioButton}
+
+  data sources, :list, default: []
+  data table_model, :struct
+  data total_count, :integer, default: 0
+  data view_type, :atom, default: :card
+  data live_action, :atom
+
+  data params, :map,
+    default: %{
+      offset: 0,
+      limit: 20,
+      sort_by: "title",
+      sort_order: "asc",
+      query: ""
+    }
+
+  prop session, :map, required: true
+  prop on_select, :event, required: true
+  prop on_select_target, :any, required: true
+  prop current_user, :map, required: true
+  prop lti_params, :map, required: true
+
+  def update(
+        %{
+          session: session,
+          on_select: on_select,
+          on_select_target: on_select_target,
+          current_user: current_user,
+          lti_params: lti_params
+        } = assigns,
+        socket
+      ) do
+    if !socket.assigns[:loaded] do
+      context = SessionContext.init(session)
+
+      # live_action is :independent_learner, :admin or :lms_instructor
+      live_action = session["live_action"]
+
+      sources =
+        retrieve_all_sources(live_action, %{user: current_user, lti_params: lti_params})
+        |> Enum.with_index(fn element, index -> Map.put(element, :unique_id, index) end)
```

Comment:
NIT: maybe this `Enum` could be merged in the previous `retrieve_all_sources` function, to increase readability?